storage/innobase/handler/handler0alter.cc

```diff
@@ -2371,23 +2362,6 @@ innobase_check_index_keys(
 				}
 			}
 
-#ifdef MYSQL_RENAME_INDEX
-			/* If a key by the same name is being created and
-			renamed, the name clash is OK. E.g.
-			ALTER TABLE t ADD INDEX i (col), RENAME INDEX i TO x
-			where the index "i" exists prior to the ALTER command.
-			In this case we:
-			1. rename the existing index from "i" to "x"
-			2. add the new index "i" */
-			for (uint i = 0; i < info->index_rename_count; i++) {
-				const KEY_PAIR*	pair
-					= &info->index_rename_buffer[i];
-
-				if (0 == strcmp(key.name, pair->old_key->name)) {
-					goto name_ok;
-				}
-			}
-#endif /* MYSQL_RENAME_INDEX */
 
 			my_error(ER_WRONG_NAME_FOR_INDEX, MYF(0),
```

Comment:
You should remove one empty line too, to not leave 2 successive empty lines inside the function body.

Is the scenario really impossible in MariaDB? I would suggest to adapt the following tests from MySQL 5.7:
```
mysql-test/suite/innodb/t/innodb_rename_index.test
mysql-test/suite/innodb/t/innodb_rename_index_err.test
```
I am not sure if the latter one is applicable. In the first one, you would obviously have to rewrite the `RENAME INDEX` into `DROP INDEX` and `ADD INDEX`. So, for example,
```sql
ALTER TABLE t ADD INDEX i(col), RENAME INDEX i TO x;
```
would become something like the following, assuming that the being-renamed index was called `i(other_col)`:
```sql
ALTER TABLE t ADD INDEX i(col), DROP INDEX i, ADD INDEX x(other_col);
```